flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/enumerator/IncrementalSourceEnumerator.java

```diff
@@ -119,6 +144,9 @@ public void handleSourceEvent(int subtaskId, SourceEvent sourceEvent) {
                     (FinishedSnapshotSplitsReportEvent) sourceEvent;
             Map<String, Offset> finishedOffsets = reportEvent.getFinishedOffsets();
             splitAssigner.onFinishedSplits(finishedOffsets);
+
+            wakeupBinlogReaderIfNeed();
```

Comment:
```suggestion
            wakeupStreamReaderIfNeed();
```

Reply:
Addressed in https://github.com/ververica/flink-cdc-connectors/pull/1838/commits/7490d9b9785adb1db065b68cd2305458e46920aa